torchgen/executorch/api/custom_ops.py

```diff
@@ -0,0 +1,163 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Sequence
+
+from torchgen import dest
+
+from torchgen.context import method_with_native_function
+
+from torchgen.gen import get_native_function_schema_registrations
+from torchgen.model import BackendIndex, DispatchKey, NativeFunction, Variant
+
+# disable import sorting to avoid circular dependency.
+from torchgen.api.types import DispatcherSignature  # isort:skip
+from torchgen.selective_build.selector import SelectiveBuilder
+from torchgen.utils import concatMap, FileManager, mapMaybe, Target
+
+# Generates RegisterKernelStub.cpp, which provides placeholder kernels for custom operators. This will be used at
+# model authoring side.
+@dataclass(frozen=True)
+class ComputeNativeFunctionStub:
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> Optional[str]:
+        if Variant.function not in f.variants:
+            return None
+
+        sig = DispatcherSignature.from_schema(
+            f.func, prefix=f"wrapper_{f.func.name.overload_name}_", symint=False
+        )
+        assert sig is not None
+        if len(f.func.returns) == 0:
+            ret_name = ""
+        elif len(f.func.returns) == 1:
+            if f.func.arguments.out:
+                ret_name = f.func.arguments.out[0].name
+            else:
+                ret_name = next(
+                    (
+                        a.name
+                        for a in f.func.arguments.flat_non_out
+                        if a.type == f.func.returns[0].type
+                    ),
+                    "",
+                )
+            if not ret_name:
+                raise Exception(f"Can't handle this return type {f.func}")
+        else:
+            assert len(f.func.arguments.out) == len(f.func.returns), (
+                "Out variant number of returns need to match the number of out arguments."
+                f" Got outs {str(f.func.arguments.out)} but returns {str(f.func.returns)}"
+            )
+            # returns a tuple of out arguments
+            tensor_type = "at::Tensor &"
+            comma = ", "
+            ret_name = f"""::std::tuple<{comma.join([tensor_type] * len(f.func.returns))}>(
+                {comma.join([r.name for r in f.func.arguments.out])}
+            )"""
+        ret_str = f"return {ret_name};" if len(f.func.returns) > 0 else ""
+        return f"""
+{sig.defn()} {{
+    {ret_str}
+}}
+    """
+
+
+def gen_custom_ops(
```

Comment:
Hmm, this doesn't follow the convention for the other api files, which is that the actual file manager heavy lifting happens at the top level executorch driver script (wherever that lives). The main reason for this is circular deps, as you have noticed when doing the imports for this file. Consider reorganizing it in that way.